### ISSUE TITLE
Mongodb connector 3.9.0 hangs on various calls for loopback persisted Models

### ISSUE THREAD

**Issue by tflowcamdev** [strongloop/loopback-connector-mongodb#479]
# Description/Steps to reproduce

The 3.9.0 version suddenly hangs for us when doing things like Model.create() or Model.upsertWithWhere().

Very simple example using a persisted model called "DataType":

```
console.log('  Calling DataType.create() ...');
  DataType.create(requestData, function(error, dtInstance) {
     // Never gets here 
     console.log('  Calling DataType.create() DONE');
     return callback(error, dtInstance);
   });
```

# Additional information

The issue appears to be related to using util.inspect with depth: null in debug statements
(e.g. here:  https://github.com/strongloop/loopback-connector-mongodb/blob/master/lib/mongodb.js#L2005)

Strongloop creates objects that have nested properties - for example a persisted model will have sub properties that link to all the models (including itself).  So traversing the DataType model properties to infinite depth results in a loop.

I don't believe we're doing anything special to produce this scenario, think this is common to strongloop's persisted models.  Using utils.inspect with a smaller depth should fix the issue.

**Comment by jannyHou** [comment]
@tflowcamdev Good catch, do you run into the same error with other models? Or only the model named `DataType`?

**Comment by epostea** [comment]
@jannyHou We hit this issue with other models, the issue is not specific to the `DataType` model

**Comment by tflowcamdev** [comment]
Yes, confirmed - This is not specific to any of our models as far as I can tell.

**Comment by nursultan156** [comment]
meet the same problem.
i call:
```
CalculationM.create(calculation, {userId: settings.userId}, (e2, r2) => {
	// this never happen
          if (e2) return e(resultFactory.internal_error(e2));
          return r(resultFactory.success(null, r2));
        });
```

memory leak occurs:
```
<--- Last few GCs --->

[1:0x41460e0]   108243 ms: Mark-sweep 1410.1 (1462.9) -> 1410.1 (1446.9) MB, 3184.4 / 0.0 ms  (+ 0.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 3184 ms) last resort GC in old space requested
[1:0x41460e0]   111318 ms: Mark-sweep 1410.1 (1446.9) -> 1410.1 (1446.9) MB, 3074.6 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1ad9c59a5ec1 <JSObject>
    1: formatValue(aka formatValue) [util.js:~392] [pc=0xe2319fe5c58](this=0x2d8304502311 <undefined>,ctx=0x32e5f7d02201 <Object map = 0x3eca1b2f9b51>,value=0x1196df2cb6d9 <JSArray[1]>,recurseTimes=0x2d8304502201 <null>,ln=0x2d83045023b1 <true>)
    2: formatProperty(aka formatProperty) [util.js:~826] [pc=0xe2319fe4d31](this=0x2d8304502311 <undefined>,ctx=0x32e5f7d02201 <Object map = 0x3eca1b2f9...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x12190dc [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewStruct(v8::internal::InstanceType) [node]
 6: v8::internal::Factory::NewTuple3(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>) [node]
 7: v8::internal::LoadIC::LoadFullChain(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Smi>) [node]
 8: v8::internal::LoadIC::UpdateCaches(v8::internal::LookupIterator*) [node]
 9: v8::internal::LoadIC::Load(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Name>) [node]
10: 0xf0c105 [node]
11: v8::internal::Runtime_KeyedLoadIC_Miss(int, v8::internal::Object**, v8::internal::Isolate*) [node]
12: 0xe2319e0463d

```

**Comment by jannyHou** [comment]
Could someone quickly hack the code, remove the `{depth: null}`, and see if that fixes the problem?

I don't have an app reproduces the error and all our existing tests pass, so I am not sure what is the use case that runs into this error, if anyone could verify ^ on their end that will save time. Thanks!

I also created the PR https://github.com/strongloop/loopback-connector-mongodb/pull/480

**Comment by tflowcamdev** [comment]
@jannyHou I've verified that commenting out those debug lines prevents the hang problem in my environment, and I've also tested doing util.inspect removing {depth: null} and that also works.

**Comment by jannyHou** [comment]
@tflowcamdev Great, thank you all for catching the bug and replying in time 👍

**Comment by dhmlau** [comment]
The fix has been released in `loopback-connector-mongodb@3.9.1`.  Thanks!

**Comment by tflowcamdev** [comment]
I confirmed version 3.9.1 fixes the problem - thanks for the quick turnaround on this one!
